File: 0242-valid-anagram/0242-valid-anagram.py
class Solution:
    def isAnagram(self, s: str, t: str) -> bool:
        if len(s) != len(t): return False
        
        countS, countT = {}, {}
        for i in s:
            countS[i] = countS.get(i, 0) + 1

        for j in t:
            countT[j] = countT.get(j, 0) + 1

        for k in countS:
            if k not in countT:return False
            if countS[k] != countT[k]: return False

        return True

File: 0242-valid-anagram/0242-valid-anagram.java
class Solution {
    public boolean isAnagram(String s, String t) {
        if(s.length() != t.length()) { return false; }
        Map<Character,Integer> countOfS = new HashMap<>();
        Map<Character,Integer> countOfT = new HashMap<>();

        for (char c: s.toCharArray()){
            countOfS.put(c,countOfS.getOrDefault(c,0) + 1);
        }
        for (char c: t.toCharArray()){
            countOfT.put(c,countOfT.getOrDefault(c,0) + 1);
        }

        for (char key : countOfS.keySet()){
            if (!countOfT.containsKey(key)) { return false; }
            if  (!countOfS.get(key).equals(countOfT.get(key))) { return false; }
        }
        return true;
        
    }
}